### ISSUE TITLE
ValueContainer loose click event after isDisabled state

### ISSUE THREAD

**Issue by pascalduez** [JedWatson/react-select#4906]
## Are you reporting a bug or runtime error?

A bug.

## Explanation
An `isDisabled` property is dynamically set then deactivated on a bare select.   
After it being deactivated, the value container (`ValueContainer`) can't receive any click events, or at least does not have any effect anymore (opening menu, etc), while the dropdown indicator still can...   
Focusing out of the select, then clicking again works as expected.   

## Reproduction
See https://codesandbox.io/s/smoosh-cloud-3mqis?file=/src/App.js

## Versions
`react-select@latest` as well as `4.x`

**Comment by Rall3n** [comment]
Hello @pascalduez,

in which browser does this problem occur?
I tried your CodeSandbox example in a Chromium browser and the problem does not occur as you describe.

**Comment by pascalduez** [comment]
Hello @Rall3n,

Ah yes, sorry for the missing curcial info :)

Reprocuded in:
`Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:94.0) Gecko/20100101 Firefox/94.0`  
`Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:96.0) Gecko/20100101 Firefox/96.0` (nightly)

**Comment by ebonow** [comment]
I am unable to reproduce this in Firefox version 94.0.1 on OSX.

Just to be clear, the reported issue is that the menu can still be opened on a disabled Select by clicking on the dropdown indicator correct... or is there some other expected behavior that you are reporting?

**Comment by pascalduez** [comment]
@ebonow Nope, the issue is once the select turns back to normal state (after being disabled), it can't be opened by clicking on the value container. But after a blur, it's back to normal...

**Comment by ebonow** [comment]
Ok got it. I was able to reproduce on Firefox. One observation about this though is that when I open up the dev tool to inspect the element, I went back into the app and it behaves normally. 

This leads me to believe it is perhaps tied to some kind of rendering issue with Firefox. There is still some jank with the styling of the input not stretching the width of the Select, so maybe that could help, but its unclear to me why the onClick handler isn't firing after the state for isDisabled is changed.

**Comment by lparolari** [comment]
I got the same issue.

I was able to reproduce it on Firefox:
* Firefox (97.0.1 64-bit), Windows 10
* Firefox (97.0.1 64-bit), Arch Linux

Please tell me if you need more information on this, I'm glad to help.

**Comment by Rall3n** [comment]
This is a problem of how browsers handle to-be-disabled elements differently.

I ran the provided CodeSandbox example in both a current Chromium browser and Firefox. I have added an interval which logs the currently focused element (`document.activeElement`) to the console. I also added handlers for `onFocus` and `onBlur` to check if any occurred

When testing the example in Firefox, I could observe that it keeps the `<input>` element as the active element when the component gets disabled, but it does not fire a `blur` event in process. The Chromium browser does not fire a `blur` event either, but it does not keep the `<input>` element as the active element (instead it moves focus to `<body>`). Also Firefox does not fire a `focus` event when re-enabling the component.

This difference is why it does not work with Firefox, but with every other browser instead. Possible fix would be to do the same we do for programmatically disabling but in reverse.
